tests_e2e/tests/lib/network_security_rule.py

```diff
@@ -134,23 +152,21 @@ def _get_network_security_group(self) -> Dict[str, Any]:
             #         }
             #     }
             #
-            subnets_copy = network_resource["properties"].get("copy") if network_resource.get("properties") is not None else None
+            subnets_copy = network_resource["properties"].get("copy") if network_resource.get(
+                "properties") is not None else None
             if subnets_copy is None:
                 raise Exception("Cannot find the copy property of the virtual network in the ARM template")
 
-            subnets = [i for i in subnets_copy if "name" in i and i["name"] == 'subnets']
-            if len(subnets) == 0:
+            subnets_input = [i for i in subnets_copy if "name" in i and i["name"] == 'subnets']
```

Comment:
why the change of name? this is not "input", is it?

Reply:
actual settings for the subnet are inside input. At this point, subnets with input property included

Reply:
maybe rename it? the name gives the impression this is the "input" property

"subnets_with_input"?